.github/workflows/build.yml

```
@@ -11,32 +11,81 @@ jobs:
         include:
           - name: Ubuntu 18.04 (gcc)
             os: ubuntu-18.04
-            cmake_args: -DFAAD=ON -DKEYFINDER=ON -DLOCALECOMPARE=ON -DMAD=ON -DMODPLUG=ON -DWAVPACK=ON
+            cmake_args: >-
+              -DFAAD=ON
+              -DKEYFINDER=ON
+              -DLOCALECOMPARE=ON
+              -DMAD=ON
+              -DMODPLUG=ON
+              -DWAVPACK=ON
+            cmake_generator: Unix Makefiles
             ctest_args:
-            ccache_path: ~/.ccache
+            compiler_cache: ccache
+            compiler_cache_path: ~/.ccache
             cpack_generator: DEB
+            buildenv_basepath: /home/runner/buildenv
             buildenv_script: tools/ubuntu_buildenv.sh
             artifacts_name: Ubuntu 18.04 DEB
             artifacts_path: cmake_build/*.deb
+            qt_qpa_platform: offscreen
           - name: Ubuntu 20.04 (gcc)
             os: ubuntu-20.04
-            cmake_args: -DFFMPEG=ON -DKEYFINDER=ON -DLOCALECOMPARE=ON -DMAD=ON -DMODPLUG=ON -DWAVPACK=ON
+            cmake_args: >-
+              -DFFMPEG=ON
+              -DKEYFINDER=ON
+              -DLOCALECOMPARE=ON
+              -DMAD=ON
+              -DMODPLUG=ON
+              -DWAVPACK=ON
+            cmake_generator: Unix Makefiles
             ctest_args:
-            ccache_path: ~/.ccache
+            compiler_cache: ccache
+            compiler_cache_path: ~/.ccache
             cpack_generator: DEB
+            buildenv_basepath: /home/runner/buildenv
             buildenv_script: tools/ubuntu_buildenv.sh
             artifacts_name: Ubuntu 20.04 DEB
             artifacts_path: cmake_build/*.deb
+            qt_qpa_platform: offscreen
           - name: macOS 10.15
             os: macos-10.15
-            cmake_args: -DCOREAUDIO=ON -DHSS1394=ON -DMACOS_BUNDLE=ON
+            cmake_args: >-
+              -DCOREAUDIO=ON
+              -DHSS1394=ON
+              -DMACOS_BUNDLE=ON
+            cmake_generator: Unix Makefiles
             # TODO: Fix this broken test on macOS
             ctest_args: --exclude-regex DirectoryDAOTest.relocateDirectory
             cpack_generator: DragNDrop
-            ccache_path: /Users/runner/Library/Caches/ccache
+            compiler_cache: ccache
+            compiler_cache_path: /Users/runner/Library/Caches/ccache
+            buildenv_basepath: /Users/runner/buildenv
             buildenv_script: tools/macos_buildenv.sh
             artifacts_name: macOS DMG
             artifacts_path: cmake_build/*.dmg
+            qt_qpa_platform: offscreen
+          - name: Windows 2016 (MSVC)
+            os: windows-2016
```

```suggestion
            os: windows-2019
```
Why an old version of Windows?

---

Is everyone using Windows 2019 yet? WIll this affect backwards-compatiblity? I don't know.

Also, the `windows-2019` image only has Visual Studio Enterprise 2019. `windows-2016` has Visual Studio Enterprise 2017 which is what we used on AppVeyor.
No idea what difference this makes..

---

One the other hand using `windows-2019` would allow use to use caching (on `windows-2016` this does not work because there is no proper `tar.exe`. By default it uses the `tar.exe` from Git Bash, which also fails because it doesn't understand windows paths.

---

I updated to windows 2019. AFAIK Windows users forced to update anyway, so let's hope this is not an issue.